pandas/tests/dtypes/test_common.py

```diff
@@ -13,6 +14,19 @@
 
 class TestPandasDtype(tm.TestCase):
 
+    # Passing invalid dtype, both as a string or object, must raise TypeError
+    def test_invalid_dtype_error(self):
+        msg = 'not understood'
+        invalid_list = [pd.Timestamp, 'pd.Timestamp']
+        for dtype in invalid_list:
+            with tm.assertRaisesRegexp(TypeError, msg):
+                pandas_dtype(dtype)
+
+        valid_list = [object, 'float64', np.object_, list,
```

Comment:
> 'list is not a valid dtype'
@jreback That would result in code such as `Series(dtype=list)` raising an error.
Among other things, this [test](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/test_strings.py#L1214) would break.